devicetypes/smartthings/zigbee-motion-sensor-light-light.src/led-cpx-light.groovy

```diff
@@ -0,0 +1,187 @@
+ *
+ *  LED CPX light
+ *
+ *  Author: SAMSUMG LED
+ *  Date: 2021-12-08
+ */
+
+import physicalgraph.zigbee.zcl.DataType
+import groovy.json.JsonOutput
+
+metadata {
+	definition(name: "LED CPX light", namespace: "SAMSUNG LED", author: "SAMSUNG LED", runLocally: true, minHubCoreVersion: '000.019.00012', executeCommandsLocally: true, genericHandler: "Zigbee") {
+		capability "Actuator"
+		capability "Color Temperature"
+		capability "Configuration"
+		capability "Health Check"
+		capability "Refresh"
+		capability "Switch"
+		capability "Switch Level"
+		capability "Light"
+		
+		attribute "colorName", "string"
+		
+		// Samsung LED
+		fingerprint profileId: "0104", inClusters: "0000, 0003, 0004, 0005, 0006, 0008, 0300, 0406", outClusters: "0019", manufacturer: "Juno", model: "ABL-LIGHTSENSOR-Z-001", deviceJoinName: "ABL CPX Light"
+	}
+  
+	// UI tile definitions
+	tiles(scale: 2) {
+		multiAttributeTile(name: "switch", type: "lighting", width: 6, height: 4, canChangeIcon: true) {
+			tileAttribute("device.switch", key: "PRIMARY_CONTROL") {
+				attributeState "on", label: '${name}', action: "switch.off", icon: "st.switches.light.on", backgroundColor: "#00A0DC", nextState: "turningOff"
+				attributeState "off", label: '${name}', action: "switch.on", icon: "st.switches.light.off", backgroundColor: "#ffffff", nextState: "turningOn"
+				attributeState "turningOn", label: '${name}', action: "switch.off", icon: "st.switches.light.on", backgroundColor: "#00A0DC", nextState: "turningOff"
+				attributeState "turningOff", label: '${name}', action: "switch.on", icon: "st.switches.light.off", backgroundColor: "#ffffff", nextState: "turningOn"
+			}
+			tileAttribute("device.level", key: "SLIDER_CONTROL") {
+				attributeState "level", action: "switch level.setLevel"
+			}
+		}
+		
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", label: "", action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+		
+		controlTile("colorTempSliderControl", "device.colorTemperature", "slider", width: 4, height: 2, inactiveLabel: false, range: "(2700..6500)") {
+			state "colorTemperature", action: "color temperature.setColorTemperature"
+		}
+		
+		valueTile("colorName", "device.colorName", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "colorName", label: '${currentValue}'
+		}
+		
+		main(["switch"])
+		details(["switch", "refresh", "switchLevel", "colorTempSliderControl", "colorName"]) //
+	}
+}
+
+// Parse incoming device messages to generate events
+def parse(String description) {
+	def event = zigbee.getEvent(description)
+	def zigbeeMap = zigbee.parseDescriptionAsMap(description)
+	if (event) {
+		if (event.name == "level" && event.value == 0) {
+		} else {
+			if (event.name == "colorTemperature") {
+				setGenericName(event.value)
+			}
+			sendEvent(event)
+		}
+	} else if (zigbeeMap.cluster == "0406" && zigbeeMap.attrId == "0000") {
```

Comment:
Can you change those "magic numbers" into constants, like it's done in another DTHes?